src/hip/handlehip.cpp

```diff
@@ -230,23 +234,43 @@ struct HandleImpl
     rocblas_handle_ptr rhandle_;
     using RocblasHandlePtrPool = std::vector<rocblas_handle_ptr>;
 #endif
+#if MIOPEN_USE_HIPBLASLT
+    hipblasLt_handle_ptr hip_blasLt_handle;
+    using HipblasLtHandlePtrPool = std::vector<hipblasLt_handle_ptr>;
```

Comment:
This is useless and wrong. If a hipblasLt_handle is created without connection to a specific stream, there is no point in producing entities

Reply:
@shurale-nkn I actually asked the hipBLASLt team about this when implementing.

Here is what I asked:

"Previously with rocBLAS, the rocBLAS handle allocated device memory that could not be shared by multiple async streams at the same time, and thus we had to keep track of which rocBLAS handle corresponded to which stream, and make a separate one for each active stream being used.
 
With hipBLASLt, it looks like that is no longer a limitation, and we can have a single hipblasLtHandle that will work for all streams. Is that accurate? Specifically, I don't see that mentioned in the documentation, and the API appears to take in streams instead of having api for setting the stream on the handle."

and here was the response:

"No, we have similar behavior. Different stream should use different handle."

I believe this means we need the above, otherwise if we use hipBLASLt in a multi-threaded situation it won't work properly.

Reply:
I couldn't find any confirmations in the documentation, but looking at the project code, they really allocate something.
It's a strange design decision to make part of the api without a handle, and make an implicit dependency between the handle and the used stream. 
but okay, we'll have to create them in the same manner.

Reply:
Yea I was surprised as well, and originally had it as just a single handle.